src/Scope.js

```diff
@@ -0,0 +1,152 @@
+import { blank, keys } from './utils/object';
+
+// A minimal `Identifier` implementation. Anything that has an `originalName`,
+// and a mutable `name` property can be used as an `Identifier`.
+function Identifier ( name ) {
```

Comment:
This is more a question of style, but should this be a `class`? I know it's essentially acting as a struct, so I could get behind a rule that says "use `function` for struct-like objects and `class` for class-like objects".